### PR TITLE
ciphey 5.14.0 (new formula)

### PR DESCRIPTION
**Pull request by alebcay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/ciphey.rb
Comment (nandahkrishna) [inline Formula/ciphey.rb:38]: Perhaps it would be better to separate this from the PyPI resources, that way it can easily be commented out to use `update-python-resources`.

**Review by nandahkrishna — DISMISSED** [review]

**Review by nandahkrishna — DISMISSED** [review]

**Comment by iMichka** [comment]
I'm working on a fix for the Linux CI.

**Comment by harens** [comment]
I added Ciphey to MacPorts and wrote a few things about packaging Ciphey (https://github.com/Ciphey/CipheyCore/issues/17). I'm just going to have a quick look through the formula before it's merged if that's ok.

Also, congrats @alebcay on getting this working!

File: Formula/ciphey.rb
Comment (harens) [inline Formula/ciphey.rb:162]: At MacPorts, we used `poetry build -f sdist --verbose` since we only require the standard distribution, and --verbose helps with debugging.

File: Formula/ciphey.rb
Comment (harens) [inline Formula/ciphey.rb:157]: I don't think this is an issue here, but just in case I thought I'd mention this just in case: https://github.com/macports/macports-ports/pull/9964. Basically, we had to set the path to the correct python version to prevent segmentation faults.

File: Formula/ciphey.rb
Comment (harens) [inline Formula/ciphey.rb:14]: Just curious, why is libyaml required?

**Review by harens — COMMENTED** [review]
Thanks @alebcay for doing this. I left a few notes that may or may not be applicable for Homebrew. Also just so you're aware, cipheycore requires C++20 if that makes any difference.

File: Formula/ciphey.rb
Comment (nandahkrishna) [reply Formula/ciphey.rb:14]: We build the `resource`s from source and if I'm not wrong PyYAML requires `libyaml` to be installed, for it to be built from source.

Comment (harens) [reply Formula/ciphey.rb:14]: Ah ok that would make sense. We had just set it as a dependency in the pyyaml port, which ciphey would itself then depend on.

Thanks @nandahkrishna for the explanation :)

File: Formula/ciphey.rb
Comment (harens) [inline Formula/ciphey.rb:131]: pywhat is [already in Homebrew](https://github.com/Homebrew/homebrew-core/blob/aea80fe8b580695c92e1978b27d6229cd8b834ed/Formula/pywhat.rb). Should this be treated in a similar way to six?

File: Formula/ciphey.rb
Comment (harens) [inline Formula/ciphey.rb:86]: Just FYI, the ciphey team can be a bit neglectful with their dependencies. For example, loguru isn't actually required, and although they removed it from their `pyproject.toml`, they didn't in the PyPi `setup.py`. I think the same is true for some other deps in the formula.

They've also added a lot of dev deps in the runtime deps section, so there's quite a few unnecessary dependencies present in this formula (e.g. flake8, isort, etc). It might be simpler just to allow them in to make the bump action simpler, but that's your call.

_(Just so you're aware, at MacPorts, I only added the necessary deps)_.

File: Formula/ciphey.rb
Comment (harens) [inline Formula/ciphey.rb:71]: Just FYI, the ciphey team have some dev deps in the runtime deps section, so there's quite a few unnecessary dependencies present in this formula (e.g. flake8, isort, pylint etc). It might be simpler just to allow them in to make the bump action simpler, but that's your call.

_(Just so you're aware, at MacPorts, I only added the necessary deps)_.

File: Formula/ciphey.rb
Comment (nandahkrishna) [reply Formula/ciphey.rb:131]: The `pywhat` formula and many other PyPI formulae are present in Homebrew as "applications" and may not install Python bindings (the "library"). The `pywhat` formula doesn't install the bindings – the formula exists as an application rather than a library, so it should be a resource.

So far, formulae have been created for frequently used packages like `numpy` and `scipy`, which install the Python bindings and are used with `depends_on` rather than being treated as resources. Given how frequently `six` is used, it was made a formula.

I'm not too familiar with the discussion around Python libraries as formulae but you may be interested in the discussion here: https://github.com/Homebrew/homebrew-core/issues/73875.

Comment (harens) [reply Formula/ciphey.rb:131]: > The pywhat formula doesn't install the bindings – the formula exists as an application rather than a library, so it should be a resource.

For some reason, I just assumed this would be the case since it's true for the other "ciphey" projects (name-that-hash, search-that-hash and this PR). Guess I was wrong `¯\_(ツ)_/¯`

Thanks again for correcting me @nandahkrishna, and thank you for not only taking the time out to write such a detailed response, but for all your work in helping to improve the Homebrew project!

**Comment by iMichka** [comment]
@alebcay if you rebase this the Linux CI should be green, we do not build Linux bottles for new formulae (yet) here. I fixed the CI for that case.

**Comment by dawidd6** [comment]
Don't worry. No need to rebase, just rerun jobs.

**Comment by alebcay** [comment]
Thanks for the pointers, @harens. I've left the dev deps in for now. Since Homebrew doesn't have a notion of formula ownership, keeping all of the deps in makes future maintenance easier for people unfamiliar with the nature of which dependencies are _actually_ needed, especially if Ciphey's dependencies change.

File: Formula/ciphey.rb
Comment (carlocab) [inline Formula/ciphey.rb:168]:
```suggestion
        venv.pip_install Dir["dist/cipheycore-*"].first
```
Doesn't this work?

**Review by harens — DISMISSED** [review]
LGTM. Thanks for submitting this @alebcay! 🎉

File: Formula/ciphey.rb
Comment (alebcay) [reply Formula/ciphey.rb]: Initially I had concerns about `venv.pip_install` treating strings differently but this seems to work.

**Review by nandahkrishna — APPROVED** [review]

File: Formula/ciphey.rb
Comment (carlocab) [reply Formula/ciphey.rb]: They should be treated the same way -- neither of them `repond_to?(:stage)`.

**Review by carlocab — APPROVED** [review]

File: Formula/ciphey.rb
Comment (alebcay) [reply Formula/ciphey.rb]: I recall there was a mode where you can pass a string containing the contents of a `requirements.txt` file but I believe the string has to be multiline for that to happen.

In any case, I find this kind of method overloading to be a bit of a weird code smell when I have to keep track of these kinds of corner cases.

**Pull request closed without merging**
